barretenberg/cpp/src/barretenberg/vm2/simulation/events/tx_events.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <cstdint>
+#include <variant>
+
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/common/field.hpp"
+
+namespace bb::avm2::simulation {
+
+struct EnqueuedCallEvent {
+    FF msg_sender;
+    FF contract_address;
+    bool is_static;
+    FF calldata_hash;
+    bool success;
+};
+
+struct PrivateAppendTreeEvent {
+    FF leaf_value;
+    uint64_t size;
+};
+
+struct PrivateEmitL2L1MessageEvent {
+    ScopedL2ToL1Message scoped_msg;
+    // TODO: Add Counter?
+};
+
+struct CollectGasFeeEvent {
+    FF fee_per_da_gas;
```

Comment:
Are these things really fields? (given that gas is u32) ugh

Reply:
sadly yes :(